src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -2655,7 +2656,7 @@ static CompositeChange clearTransportedByForAlliedAirOnCarrier(final Collection<
     if (!carriers.isEmpty() && !games.strategy.triplea.Properties.getAlliedAirIndependent(data)) {
       final Match<Unit> alliedFighters = new CompositeMatchAnd<>(Matches.isUnitAllied(attacker, data),
           Matches.unitIsOwnedBy(attacker).invert(), Matches.UnitIsAir, Matches.UnitCanLandOnCarrier);
-      final Collection<Unit> alliedAirInTerr = Match.getMatches(battleSite.getUnits().getUnits(), alliedFighters);
+      final Collection<Unit> alliedAirInTerr = Match.getMatches(attackingUnits, alliedFighters);
```

Comment:
I think you are right, but I can't remember if defending units need to be cleared as well or not.  Can you check that?